iis-wf/iis-wf-documentsclassification/src/main/java/eu/dnetlib/iis/wf/documentsclassification/DocumentClassificationJob.java

```diff
@@ -51,10 +53,15 @@ public static void main(String[] args) throws IOException {
 
             sc.sc().addFile(params.scriptDirPath, true);
 
-            JavaRDD<ExtractedDocumentMetadataMergedWithOriginal> documents = avroLoader
+            JavaRDD<ExtractedDocumentMetadataMergedWithOriginal> rawDocuments = avroLoader
                     .loadJavaRDD(sc, params.inputAvroPath, ExtractedDocumentMetadataMergedWithOriginal.class);
             
-            JavaRDD<DocumentMetadata> metadataRecords = documents
+            JavaRDD<ExtractedDocumentMetadataMergedWithOriginal> repartDocuments = 
+                    (StringUtils.isNotBlank(params.numberOfPartitions) && !WorkflowRuntimeParameters.UNDEFINED_NONEMPTY_VALUE.equals(params.numberOfPartitions))
```

Comment:
A method could be introduced to encapsulate repartitioning condition e.g. `shouldRepartition(...)`.